staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4996,11 +4996,21 @@ type EndpointPort struct {
 	// +optional
 	Protocol Protocol `json:"protocol,omitempty" protobuf:"bytes,3,opt,name=protocol,casttype=Protocol"`
 
-	// The application protocol for this port.
+	// Used as a hint for implementations to
+	// configure the protocol used between the
+	// implementation and the application it exposes.
 	// This field follows standard Kubernetes label syntax.
-	// Un-prefixed names are reserved for IANA standard service names (as per
+	// Valid values are either:
+	//
+	// * Un-prefixed protocol names - reserved for IANA standard service names (as per
 	// RFC-6335 and https://www.iana.org/assignments/service-names).
-	// Non-standard protocols should use prefixed names such as
+	//
+	// * Kubernetes standard names:
+	//   * 'k8s.io/h2c' - http2 over cleartext as described in https://www.rfc-editor.org/rfc/rfc7540
+	//   * 'k8s.io/grpc' - grpc traffic - see https://github.com/grpc/grpc/blob/v1.51.1/doc/PROTOCOL-HTTP2.md
+	//   * 'k8s.io/tcp' - plain tcp traffic
+	//
+	// * Other protocols should use prefixed names such as
```

Comment:
"user-defined prefixed names"

Reply:
I'd say it is more of "implementation-defined" than user defined. objections ?

Reply:
I'm fine with that